docs/content/guides/dagster/branch_depl_with_snowflake.mdx

```diff
@@ -0,0 +1,325 @@
+    return result
+
+
+@asset
+def comments(items: pd.DataFrame) -> pd.DataFrame:
+    """Comments from the Hacker News API."""
+    return items[items["type"] == "comment"]
+
+
+@asset
+def stories(items: pd.DataFrame) -> pd.DataFrame:
+    """Stories from the Hacker News API."""
+    return items[items["type"] == "story"]
+```
+
+As you can see, our assets use a [resource](https://docs.dagster.io/\_apidocs/libraries/dagster-snowflake#dagster_snowflake.snowflake_resource) named `hn_client` (source code available in the [Transitioning Data Pipelines from Development to Production](/guides/dagster/local-to-production-tutorial#part-one-local-development) guide). Using a resource allows us to swap out implementations per environment without modifying our computational code.
+
+### Step 2: Configure our job for each environment
+
+At runtime, we’d like to determine which environment our code is running in: branch deployment, or production. This is valuable information that will dictate how our code should execute—with which credentials and with which database. We’ll use a different set of credentials in each environment to ensure we can’t accidentally write to production within our branch deployment, and we’ll write to our database clone in our branch deployment instead of `PRODUCTION`.
+
+Dagster automatically sets certain environment variables containing deployment metadata, allowing us to read these environment variables to discern between deployments. For a full list of preset environment variables, visit this link. We can access the `DAGSTER_CLOUD_IS_BRANCH_DEPLOYMENT` environment variable to determine the currently executing environment.
+
+Because we want to configure our assets to write to Snowflake using a different set of credentials and database in each environment, we’ll configure a separate IO manager for each environment:
+
+```python file=/guides/dagster/local_to_prod_tutorial/branch_depl_tutorial/repository_v1.py startafter=start_repository endbefore=end_repository
+@repository
+def repo():
+    snowflake_config = {
+        "account": {"env": "SNOWFLAKE_ACCOUNT"},
+        "user": {"env": "SNOWFLAKE_USER"},
+        "password": {"env": "SNOWFLAKE_PASSWORD"},
+        "schema": {"env": "SNOWFLAKE_SCHEMA"},
+    }
+    resource_defs = {
+        "branch": {
+            "hn_client": hn_api_client,
+            "snowflake_io_mgr": snowflake_io_manager.configured(
+                {
+                    **snowflake_config,
+                    "database": f"PRODUCTION_CLONE_{os.getenv('DAGSTER_CLOUD_PULL_REQUEST_ID')}",
+                }
+            ),
+        },
+        "production": {
+            "hn_client": hn_api_client,
+            "snowflake_io_mgr": snowflake_io_manager.configured(
+                {
+                    **snowflake_config,
+                    "database": "PRODUCTION",
+                }
+            ),
+        },
+    }
+
+    def get_current_env():
```

Comment:
One thing i'm still unclear on is what env vars are set by us in dagster cloud. In my guide I'm referencing a var called DAGSTER_DEPLOYMENT as a placeholder, but if all we have access to is a var that tells us if we're in a branch deployment or not, I'll need to come up with a way to differentiate between local and prod. do you know if there's a list of the env vars we have anywhere?

Reply:
here's a list from the PR: https://github.com/dagster-io/internal/pull/2697/files

Though this PR has since been reverted, so we'll have to double check on this once the env vars are merged again